From 66b6806a9ae2bcc0439018dc3cab13adf5cb47bb Mon Sep 17 00:00:00 2001
From: Jai Moraes <jmoraes@blackberry.com>
Date: Fri, 28 Feb 2025 15:22:59 -0500
Subject: [PATCH 1/6] QNX related changes to commandline and filesystem utils

Imported from qnx-ports/EmulationStation (51b6f4162d8e...qnx_master); original authorship kept.
Upstream-Status: QNX-specific (carried in the qnx-ports fork)


---
 CMakeLists.txt                       | 17 ++++++++++++++---
 es-app/src/ScraperCmdLine.cpp        |  2 +-
 es-core/src/utils/FileSystemUtil.cpp |  9 +++++++++
 3 files changed, 24 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4bc0e45e3b..64f34b1334 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,6 +96,14 @@ else()
     endif()
 endif()
 
+if(QNX)
+    set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+    set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)  
+    set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+    set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM ONLY)
+    set(PKG_CONFIG_EXECUTABLE "/usr/bin/pkgconf")
+endif()  
+
 find_package(Freetype REQUIRED)
 find_package(FreeImage REQUIRED)
 find_package(SDL2 REQUIRED)
@@ -108,6 +116,7 @@ if(CEC)
     find_package(libCEC REQUIRED)
 endif()
 
+#Note for QNX: Will eventually need QSA
 #add ALSA for Linux
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     find_package(ALSA REQUIRED)
@@ -155,9 +164,11 @@ endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX)
     #check for G++ 4.7+
-    execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE G++_VERSION)
-    if (G++_VERSION VERSION_LESS 4.7)
-            message(SEND_ERROR "You need at least G++ 4.7 to compile EmulationStation!")
+    if(NOT QNX) #qcc/q++ are all above v4.7, however they are missing -dumpversion
+        execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE G++_VERSION)
+        if (G++_VERSION VERSION_LESS 4.7)
+                message(SEND_ERROR "You need at least G++ 4.7 to compile EmulationStation!")
+        endif()
     endif()
 
     #set up compiler flags for GCC
diff --git a/es-app/src/ScraperCmdLine.cpp b/es-app/src/ScraperCmdLine.cpp
index dda5635957..3748d22207 100644
--- a/es-app/src/ScraperCmdLine.cpp
+++ b/es-app/src/ScraperCmdLine.cpp
@@ -5,7 +5,7 @@
 #include "SystemData.h"
 #include <iostream>
 #include <signal.h>
-#if defined(__linux__)
+#if defined(__linux__) || defined(__QNX__)
 #include <unistd.h>
 #elif defined(WIN32)
 #include <Windows.h>
diff --git a/es-core/src/utils/FileSystemUtil.cpp b/es-core/src/utils/FileSystemUtil.cpp
index 94be3a58bd..085c77685e 100644
--- a/es-core/src/utils/FileSystemUtil.cpp
+++ b/es-core/src/utils/FileSystemUtil.cpp
@@ -23,6 +23,15 @@
 #include <unistd.h>
 #endif // _WIN32
 
+#if defined(__QNX__)
+//Silly workaround, as -D_LARGEFILE64_SOURCE is not working for me.
+#define stat64 stat
+int lstat64(const char* path, struct stat64 *buf){
+	return lstat(path, buf);
+}
+#endif
+
+
 //////////////////////////////////////////////////////////////////////////
 
 namespace Utils

From c1f7ae3cd84ae7f703b07a8fcc21d3a544e4d0f4 Mon Sep 17 00:00:00 2001
From: Jai Moraes <jmoraes@blackberry.com>
Date: Wed, 12 Mar 2025 16:34:20 -0400
Subject: [PATCH 2/6] CMake QNX Update - pushed for testing

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64f34b1334..c859245eb0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -101,9 +101,9 @@ if(QNX)
     set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)  
     set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
     set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM ONLY)
-    set(PKG_CONFIG_EXECUTABLE "/usr/bin/pkgconf")
 endif()  
 
+
 find_package(Freetype REQUIRED)
 find_package(FreeImage REQUIRED)
 find_package(SDL2 REQUIRED)

From 9ba7efb64a6ea8cdddf1b5c9d55cd2b94aa83ef4 Mon Sep 17 00:00:00 2001
From: Jai Moraes <jmoraes@blackberry.com>
Date: Thu, 13 Mar 2025 10:19:18 -0400
Subject: [PATCH 3/6] Push for testing - temporary fixes combined with other
 work.

---
 es-app/src/main.cpp                 | 6 ++++++
 es-app/src/views/SystemView.cpp     | 7 +++++--
 es-app/src/views/ViewController.cpp | 5 +++++
 es-core/src/Window.cpp              | 7 +++++++
 es-core/src/guis/GuiMsgBox.cpp      | 3 +++
 es-core/src/renderers/Renderer.cpp  | 8 ++++++--
 6 files changed, 32 insertions(+), 4 deletions(-)

diff --git a/es-app/src/main.cpp b/es-app/src/main.cpp
index eb0b92cacf..93e2588b49 100644
--- a/es-app/src/main.cpp
+++ b/es-app/src/main.cpp
@@ -291,6 +291,7 @@ void onExit()
 
 int main(int argc, char* argv[])
 {
+	printf("MAIN FUNC BEGIN\n");
 	std::locale::global(std::locale("C"));
 
 	if(!parseArgs(argc, argv))
@@ -353,6 +354,7 @@ int main(int argc, char* argv[])
 	CollectionSystemManager::init(&window);
 	MameNames::init();
 	window.pushGui(ViewController::get());
+	printf("QNX: Pushing GUI L356\n"); //QNX
 
 	bool splashScreen = Settings::getInstance()->getBool("SplashScreen");
 
@@ -374,6 +376,7 @@ int main(int argc, char* argv[])
 	const char* errorMsg = NULL;
 	if(!loadSystemConfigFile(splashScreen ? &window : nullptr, &errorMsg))
 	{
+		printf("Test 1\n"); //QNX
 		// something went terribly wrong
 		if(errorMsg == NULL)
 		{
@@ -382,6 +385,7 @@ int main(int argc, char* argv[])
 				Renderer::deinit();
 			return 1;
 		}
+		printf("Test 1a\n");//QNX
 
 		// we can't handle es_systems.cfg file problems inside ES itself, so display the error message then quit
 		window.pushGui(new GuiMsgBox(&window,
@@ -392,6 +396,7 @@ int main(int argc, char* argv[])
 				SDL_PushEvent(quit);
 			}));
 	}
+	printf("Test 2\n"); //QNX
 
 	//run the command line scraper then quit
 	if(scrape_cmdline)
@@ -415,6 +420,7 @@ int main(int argc, char* argv[])
 		{
 			ViewController::get()->goToStart();
 		}else{
+			printf("QNX: Pushing GUI L422\n");
 			window.pushGui(new GuiDetectDevice(&window, true, [] { ViewController::get()->goToStart(); }));
 		}
 	}
diff --git a/es-app/src/views/SystemView.cpp b/es-app/src/views/SystemView.cpp
index d01c432b27..425cb009ec 100644
--- a/es-app/src/views/SystemView.cpp
+++ b/es-app/src/views/SystemView.cpp
@@ -344,9 +344,12 @@ void SystemView::onCursorChanged(const CursorState& /*state*/)
 }
 
 void SystemView::render(const Transform4x4f& parentTrans)
-{
-	if(size() == 0)
+{	
+	printf("SystemView render call\n");
+	if(size() == 0){
+		printf("systemview rendering exit\n");
 		return;  // nothing to render
+	}
 
 	Transform4x4f trans = getTransform() * parentTrans;
 
diff --git a/es-app/src/views/ViewController.cpp b/es-app/src/views/ViewController.cpp
index 3d5079acec..01b78009b9 100644
--- a/es-app/src/views/ViewController.cpp
+++ b/es-app/src/views/ViewController.cpp
@@ -414,6 +414,7 @@ std::shared_ptr<IGameListView> ViewController::getGameListView(SystemData* syste
 
 std::shared_ptr<SystemView> ViewController::getSystemListView()
 {
+	//printf("System View Debug\n"); //QNX
 	//if we already made one, return that one
 	if(mSystemListView)
 		return mSystemListView;
@@ -461,6 +462,7 @@ void ViewController::update(int deltaTime)
 
 void ViewController::render(const Transform4x4f& parentTrans)
 {
+	printf("Rendering Debug Message\n");
 	Transform4x4f trans = mCamera * parentTrans;
 	Transform4x4f transInverse;
 	transInverse.invert(trans);
@@ -478,6 +480,7 @@ void ViewController::render(const Transform4x4f& parentTrans)
 	// draw gamelists
 	for(auto it = mGameListViews.cbegin(); it != mGameListViews.cend(); it++)
 	{
+		printf("Gamelist count\n"); //QNX
 		// clipping
 		Vector3f guiStart = it->second->getPosition();
 		Vector3f guiEnd = it->second->getPosition() + Vector3f(it->second->getSize().x(), it->second->getSize().y(), 0);
@@ -487,12 +490,14 @@ void ViewController::render(const Transform4x4f& parentTrans)
 			it->second->render(trans);
 	}
 
+	printf("Post Gamelist count\n"); //QNX
 	if(mWindow->peekGui() == this)
 		mWindow->renderHelpPromptsEarly();
 
 	// fade out
 	if(mFadeOpacity)
 	{
+		printf("Fade\n"); //QNX
 		unsigned int fadeColor = 0x00000000 | (unsigned char)(mFadeOpacity * 255);
 		Renderer::setMatrix(parentTrans);
 		Renderer::drawRect(0.0f, 0.0f, Renderer::getScreenWidth(), Renderer::getScreenHeight(), fadeColor, fadeColor);
diff --git a/es-core/src/Window.cpp b/es-core/src/Window.cpp
index 42c9bfa746..35bbbd2bd8 100644
--- a/es-core/src/Window.cpp
+++ b/es-core/src/Window.cpp
@@ -33,13 +33,20 @@ Window::~Window()
 
 void Window::pushGui(GuiComponent* gui)
 {
+	printf("Inside Window::pushGui <<<\n"); //QNX
 	if (mGuiStack.size() > 0)
 	{
 		auto& top = mGuiStack.back();
 		top->topWindow(false);
 	}
+	printf("WPG 1, gui=%u %d\n", gui, gui);//QNX
+	if(!gui) printf("invalid gui\n");
 	mGuiStack.push_back(gui);
+	
+	printf("WPG 1.1\n");//QNX
 	gui->updateHelpPrompts();
+	printf("WPG 2\n");//QNX
+
 }
 
 void Window::removeGui(GuiComponent* gui)
diff --git a/es-core/src/guis/GuiMsgBox.cpp b/es-core/src/guis/GuiMsgBox.cpp
index 9db9ffff11..dd7317e1e3 100644
--- a/es-core/src/guis/GuiMsgBox.cpp
+++ b/es-core/src/guis/GuiMsgBox.cpp
@@ -61,6 +61,8 @@ GuiMsgBox::GuiMsgBox(Window* window, const std::string& text,
 
 	addChild(&mBackground);
 	addChild(&mGrid);
+	printf("GUIBox End\n"); //QNX
+	printf("text: %s\n", text);
 }
 
 bool GuiMsgBox::input(InputConfig* config, Input input)
@@ -96,6 +98,7 @@ void GuiMsgBox::onSizeChanged()
 
 void GuiMsgBox::deleteMeAndCall(const std::function<void()>& func)
 {
+	printf("GUI MSG BOX Delete\n");
 	auto funcCopy = func;
 	delete this;
 
diff --git a/es-core/src/renderers/Renderer.cpp b/es-core/src/renderers/Renderer.cpp
index e12a3b6418..61688f0064 100644
--- a/es-core/src/renderers/Renderer.cpp
+++ b/es-core/src/renderers/Renderer.cpp
@@ -260,10 +260,14 @@ namespace Renderer
 
 //////////////////////////////////////////////////////////////////////////
 
-	void drawRect(const float _x, const float _y, const float _w, const float _h, const unsigned int _color, const unsigned int _colorEnd, bool horizontalGradient, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
+	void drawRect(const float _x, const float _y, /*const*/ float _w, /*const*/ float _h, const unsigned int _color, const unsigned int _colorEnd, bool horizontalGradient, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
 	{
-		const unsigned int color    = convertColor(_color);
+		
+		_w = 1920;
+		_h = 1080;
+		const unsigned int color    = /*convertColor(_color);*/ convertColor(0xAABBCCFF); //QNX
 		const unsigned int colorEnd = convertColor(_colorEnd);
+		printf("Draw rect call x%f y%f w%f h%f c=0x%08X \n", _x, _y, _w, _h, color); //QNX
 		Vertex             vertices[4];
 
 		vertices[0] = { { _x     ,_y      }, { 0.0f, 0.0f }, color };

From a44b69d8d7ba0dd151db3a7f9384ea9a49f21c37 Mon Sep 17 00:00:00 2001
From: Jai Moraes <jmoraes@blackberry.com>
Date: Fri, 14 Mar 2025 14:48:13 -0400
Subject: [PATCH 4/6] Working

---
 es-app/src/main.cpp                       |  7 ++---
 es-app/src/views/SystemView.cpp           |  4 +--
 es-app/src/views/ViewController.cpp       |  6 ++--
 es-core/src/Settings.cpp                  |  8 ++---
 es-core/src/Window.cpp                    |  8 ++---
 es-core/src/renderers/Renderer.cpp        | 14 +++++++--
 es-core/src/renderers/Renderer_GL14.cpp   |  1 +
 es-core/src/renderers/Renderer_GL21.cpp   |  1 +
 es-core/src/renderers/Renderer_GLES10.cpp |  1 +
 es-core/src/renderers/Renderer_GLES20.cpp | 38 +++++++++++++++++++----
 10 files changed, 61 insertions(+), 27 deletions(-)

diff --git a/es-app/src/main.cpp b/es-app/src/main.cpp
index 93e2588b49..0d090fdfea 100644
--- a/es-app/src/main.cpp
+++ b/es-app/src/main.cpp
@@ -354,7 +354,7 @@ int main(int argc, char* argv[])
 	CollectionSystemManager::init(&window);
 	MameNames::init();
 	window.pushGui(ViewController::get());
-	printf("QNX: Pushing GUI L356\n"); //QNX
+	printf("main.cpp | Pushing GUI line 357\n"); //QNX
 
 	bool splashScreen = Settings::getInstance()->getBool("SplashScreen");
 
@@ -376,7 +376,6 @@ int main(int argc, char* argv[])
 	const char* errorMsg = NULL;
 	if(!loadSystemConfigFile(splashScreen ? &window : nullptr, &errorMsg))
 	{
-		printf("Test 1\n"); //QNX
 		// something went terribly wrong
 		if(errorMsg == NULL)
 		{
@@ -385,7 +384,6 @@ int main(int argc, char* argv[])
 				Renderer::deinit();
 			return 1;
 		}
-		printf("Test 1a\n");//QNX
 
 		// we can't handle es_systems.cfg file problems inside ES itself, so display the error message then quit
 		window.pushGui(new GuiMsgBox(&window,
@@ -396,7 +394,6 @@ int main(int argc, char* argv[])
 				SDL_PushEvent(quit);
 			}));
 	}
-	printf("Test 2\n"); //QNX
 
 	//run the command line scraper then quit
 	if(scrape_cmdline)
@@ -420,7 +417,7 @@ int main(int argc, char* argv[])
 		{
 			ViewController::get()->goToStart();
 		}else{
-			printf("QNX: Pushing GUI L422\n");
+			printf("main.cpp | Pushing GUI line 423\n");
 			window.pushGui(new GuiDetectDevice(&window, true, [] { ViewController::get()->goToStart(); }));
 		}
 	}
diff --git a/es-app/src/views/SystemView.cpp b/es-app/src/views/SystemView.cpp
index 425cb009ec..8c3cf9b03a 100644
--- a/es-app/src/views/SystemView.cpp
+++ b/es-app/src/views/SystemView.cpp
@@ -345,9 +345,9 @@ void SystemView::onCursorChanged(const CursorState& /*state*/)
 
 void SystemView::render(const Transform4x4f& parentTrans)
 {	
-	printf("SystemView render call\n");
+	printf("SystemView.cpp | SystemView render call\n");
 	if(size() == 0){
-		printf("systemview rendering exit\n");
+		printf("SystemView.cpp | systemview rendering exit due to failure\n");
 		return;  // nothing to render
 	}
 
diff --git a/es-app/src/views/ViewController.cpp b/es-app/src/views/ViewController.cpp
index 01b78009b9..15f264ea64 100644
--- a/es-app/src/views/ViewController.cpp
+++ b/es-app/src/views/ViewController.cpp
@@ -462,7 +462,7 @@ void ViewController::update(int deltaTime)
 
 void ViewController::render(const Transform4x4f& parentTrans)
 {
-	printf("Rendering Debug Message\n");
+	printf("ViewController.cpp | ::render\n");
 	Transform4x4f trans = mCamera * parentTrans;
 	Transform4x4f transInverse;
 	transInverse.invert(trans);
@@ -480,7 +480,7 @@ void ViewController::render(const Transform4x4f& parentTrans)
 	// draw gamelists
 	for(auto it = mGameListViews.cbegin(); it != mGameListViews.cend(); it++)
 	{
-		printf("Gamelist count\n"); //QNX
+		//printf("Gamelist count\n"); //QNX
 		// clipping
 		Vector3f guiStart = it->second->getPosition();
 		Vector3f guiEnd = it->second->getPosition() + Vector3f(it->second->getSize().x(), it->second->getSize().y(), 0);
@@ -490,7 +490,7 @@ void ViewController::render(const Transform4x4f& parentTrans)
 			it->second->render(trans);
 	}
 
-	printf("Post Gamelist count\n"); //QNX
+	//printf("Post Gamelist count\n"); //QNX
 	if(mWindow->peekGui() == this)
 		mWindow->renderHelpPromptsEarly();
 
diff --git a/es-core/src/Settings.cpp b/es-core/src/Settings.cpp
index aeb1057e33..375e99cbd8 100644
--- a/es-core/src/Settings.cpp
+++ b/es-core/src/Settings.cpp
@@ -172,10 +172,10 @@ void Settings::setDefaults()
 	mBoolMap["ForceKid"] = false;
 	mBoolMap["ForceDisableFilters"] = false;
 
-	mIntMap["WindowWidth"]   = 0;
-	mIntMap["WindowHeight"]  = 0;
-	mIntMap["ScreenWidth"]   = 0;
-	mIntMap["ScreenHeight"]  = 0;
+	mIntMap["WindowWidth"]   = 1920;
+	mIntMap["WindowHeight"]  = 1080;
+	mIntMap["ScreenWidth"]   = 1920;
+	mIntMap["ScreenHeight"]  = 1080;
 	mIntMap["ScreenOffsetX"] = 0;
 	mIntMap["ScreenOffsetY"] = 0;
 	mIntMap["ScreenRotate"]  = 0;
diff --git a/es-core/src/Window.cpp b/es-core/src/Window.cpp
index 35bbbd2bd8..8b9c72f317 100644
--- a/es-core/src/Window.cpp
+++ b/es-core/src/Window.cpp
@@ -33,19 +33,17 @@ Window::~Window()
 
 void Window::pushGui(GuiComponent* gui)
 {
-	printf("Inside Window::pushGui <<<\n"); //QNX
+	printf("Window.cpp: | Inside Window::pushGui <<<\n"); //QNX
 	if (mGuiStack.size() > 0)
 	{
 		auto& top = mGuiStack.back();
 		top->topWindow(false);
 	}
-	printf("WPG 1, gui=%u %d\n", gui, gui);//QNX
+	
 	if(!gui) printf("invalid gui\n");
 	mGuiStack.push_back(gui);
-	
-	printf("WPG 1.1\n");//QNX
+
 	gui->updateHelpPrompts();
-	printf("WPG 2\n");//QNX
 
 }
 
diff --git a/es-core/src/renderers/Renderer.cpp b/es-core/src/renderers/Renderer.cpp
index 61688f0064..f469404d9a 100644
--- a/es-core/src/renderers/Renderer.cpp
+++ b/es-core/src/renderers/Renderer.cpp
@@ -90,6 +90,15 @@ namespace Renderer
 		screenOffsetX = Settings::getInstance()->getInt("ScreenOffsetX") ? Settings::getInstance()->getInt("ScreenOffsetX") : 0;
 		screenOffsetY = Settings::getInstance()->getInt("ScreenOffsetY") ? Settings::getInstance()->getInt("ScreenOffsetY") : 0;
 		screenRotate  = Settings::getInstance()->getInt("ScreenRotate")  ? Settings::getInstance()->getInt("ScreenRotate")  : 0;
+		printf("Renderer.cpp | Reading in values... \n ");
+		printf("             | windowWidth %d \n ", windowWidth);
+		printf("             | windowHeight %d \n ", windowHeight);
+		printf("             | screenWidth %d \n ", screenWidth);
+		printf("             | screenHeight %d \n ", screenHeight);
+		printf("             | screenOffsetX %d \n ", screenOffsetX);
+		printf("             | screenOffsetY %d \n ", screenOffsetY);
+		printf("             | screenRotate %d \n ", screenRotate);
+		
 
 		setupWindow();
 
@@ -263,8 +272,9 @@ namespace Renderer
 	void drawRect(const float _x, const float _y, /*const*/ float _w, /*const*/ float _h, const unsigned int _color, const unsigned int _colorEnd, bool horizontalGradient, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
 	{
 		
-		_w = 1920;
-		_h = 1080;
+		//QNX
+		// _w = 1920;
+		// _h = 1080;
 		const unsigned int color    = /*convertColor(_color);*/ convertColor(0xAABBCCFF); //QNX
 		const unsigned int colorEnd = convertColor(_colorEnd);
 		printf("Draw rect call x%f y%f w%f h%f c=0x%08X \n", _x, _y, _w, _h, color); //QNX
diff --git a/es-core/src/renderers/Renderer_GL14.cpp b/es-core/src/renderers/Renderer_GL14.cpp
index 6550b4d272..dd0885df8e 100644
--- a/es-core/src/renderers/Renderer_GL14.cpp
+++ b/es-core/src/renderers/Renderer_GL14.cpp
@@ -215,6 +215,7 @@ namespace Renderer
 
 	void drawTriangleStrips(const Vertex* _vertices, const unsigned int _numVertices, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
 	{
+		printf("QNX DEBUG: GL14_DRTRISTRIPS\n"); //QNX
 		GL_CHECK_ERROR(glVertexPointer(  2, GL_FLOAT,         sizeof(Vertex), &_vertices[0].pos));
 		GL_CHECK_ERROR(glTexCoordPointer(2, GL_FLOAT,         sizeof(Vertex), &_vertices[0].tex));
 		GL_CHECK_ERROR(glColorPointer(   4, GL_UNSIGNED_BYTE, sizeof(Vertex), &_vertices[0].col));
diff --git a/es-core/src/renderers/Renderer_GL21.cpp b/es-core/src/renderers/Renderer_GL21.cpp
index 009501df12..e6e0f21240 100644
--- a/es-core/src/renderers/Renderer_GL21.cpp
+++ b/es-core/src/renderers/Renderer_GL21.cpp
@@ -215,6 +215,7 @@ namespace Renderer
 
 	void drawTriangleStrips(const Vertex* _vertices, const unsigned int _numVertices, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
 	{
+		printf("QNX DEBUG: GL21_DRTRISTRIPS\n"); //QNX
 		GL_CHECK_ERROR(glVertexPointer(  2, GL_FLOAT,         sizeof(Vertex), &_vertices[0].pos));
 		GL_CHECK_ERROR(glTexCoordPointer(2, GL_FLOAT,         sizeof(Vertex), &_vertices[0].tex));
 		GL_CHECK_ERROR(glColorPointer(   4, GL_UNSIGNED_BYTE, sizeof(Vertex), &_vertices[0].col));
diff --git a/es-core/src/renderers/Renderer_GLES10.cpp b/es-core/src/renderers/Renderer_GLES10.cpp
index 33c6e0547e..63b159352a 100644
--- a/es-core/src/renderers/Renderer_GLES10.cpp
+++ b/es-core/src/renderers/Renderer_GLES10.cpp
@@ -215,6 +215,7 @@ namespace Renderer
 
 	void drawTriangleStrips(const Vertex* _vertices, const unsigned int _numVertices, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
 	{
+		printf("QNX DEBUG: GLES10_DRTRISTRIPS\n"); //QNX
 		GL_CHECK_ERROR(glVertexPointer(  2, GL_FLOAT,         sizeof(Vertex), &_vertices[0].pos));
 		GL_CHECK_ERROR(glTexCoordPointer(2, GL_FLOAT,         sizeof(Vertex), &_vertices[0].tex));
 		GL_CHECK_ERROR(glColorPointer(   4, GL_UNSIGNED_BYTE, sizeof(Vertex), &_vertices[0].col));
diff --git a/es-core/src/renderers/Renderer_GLES20.cpp b/es-core/src/renderers/Renderer_GLES20.cpp
index 45abb22b9e..3e70bc9f90 100644
--- a/es-core/src/renderers/Renderer_GLES20.cpp
+++ b/es-core/src/renderers/Renderer_GLES20.cpp
@@ -12,8 +12,8 @@
 
 namespace Renderer
 {
-
-#if defined(_DEBUG)
+// #define _DEBUG
+// #if defined(_DEBUG)
 #define GL_CHECK_ERROR(Function) (Function, _GLCheckError(#Function))
 
 	static void _GLCheckError(const char* _funcName)
@@ -23,9 +23,10 @@ namespace Renderer
 		if(errorCode != GL_NO_ERROR)
 			LOG(LogError) << "GL error: " << _funcName << " failed with error code: " << errorCode;
 	}
-#else
-#define GL_CHECK_ERROR(Function) (Function)
-#endif
+// #else
+// #define GL_CHECK_ERROR(Function) (Function)
+// #endif
+// #undef _DEBUG
 
 //////////////////////////////////////////////////////////////////////////
 
@@ -44,6 +45,7 @@ namespace Renderer
 
 	static void setupShaders()
 	{
+		//printf("SETUP SHADERS\n");//QNX
 		// vertex shader
 		const GLchar* vertexSource =
 			"uniform   mat4 u_mvp; \n"
@@ -63,6 +65,7 @@ namespace Renderer
 		GL_CHECK_ERROR(glShaderSource(vertexShader, 1, &vertexSource, nullptr));
 		GL_CHECK_ERROR(glCompileShader(vertexShader));
 
+		printf("EmulationStation: | setupShaders\n");//QNX
 		{
 			GLint isCompiled = GL_FALSE;
 			GLint maxLength  = 0;
@@ -190,6 +193,7 @@ namespace Renderer
 
 	static void setupVertexBuffer()
 	{
+		//printf("SETUP VERTEX BUFFER\n");//QNX
 		GL_CHECK_ERROR(glGenBuffers(1, &vertexBuffer));
 		GL_CHECK_ERROR(glBindBuffer(GL_ARRAY_BUFFER, vertexBuffer));
 
@@ -199,6 +203,7 @@ namespace Renderer
 
 	static GLenum convertBlendFactor(const Blend::Factor _blendFactor)
 	{
+		//printf("CONVERT BLEND FACTOR\n");//QNX
 		switch(_blendFactor)
 		{
 			case Blend::ZERO:                { return GL_ZERO;                } break;
@@ -220,6 +225,7 @@ namespace Renderer
 
 	static GLenum convertTextureType(const Texture::Type _type)
 	{
+		//printf("CONVERT TEXTURE TYPE\n");//QNX
 		switch(_type)
 		{
 			case Texture::RGBA:  { return GL_RGBA;            } break;
@@ -239,7 +245,11 @@ namespace Renderer
 		const unsigned char b = ((_color & 0x0000ff00) >>  8) & 255;
 		const unsigned char a = ((_color & 0x000000ff)      ) & 255;
 
+#ifdef __QNX__ //SHOULD BE RGBX
+		return ((r << 24) | (g << 16) | (b << 8) | (a));
+#else
 		return ((a << 24) | (b << 16) | (g << 8) | (r));
+	#endif
 
 	} // convertColor
 
@@ -247,7 +257,8 @@ namespace Renderer
 
 	unsigned int getWindowFlags()
 	{
-		return SDL_WINDOW_OPENGL;
+		printf("EmulationStation: | WINDOW FLAGS\n");//QNX
+		return SDL_WINDOW_OPENGL; //is this right????? OPENGLES??
 
 	} // getWindowFlags
 
@@ -255,6 +266,7 @@ namespace Renderer
 
 	void setupWindow()
 	{
+		printf("EmulationStaion: | SETUP WINDOW\n");//QNX
 		SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK,  SDL_GL_CONTEXT_PROFILE_ES);
 		SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 2);
 		SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 0);
@@ -272,6 +284,7 @@ namespace Renderer
 
 	void createContext()
 	{
+		printf("EmulationStation: | CREATE CONTEXT\n");//QNX
 		sdlContext = SDL_GL_CreateContext(getSDLWindow());
 		SDL_GL_MakeCurrent(getSDLWindow(), sdlContext);
 
@@ -304,6 +317,7 @@ namespace Renderer
 
 	void destroyContext()
 	{
+		printf("EmulationStation: | DESTROY CONTEXT\n");//QNX
 		SDL_GL_DeleteContext(sdlContext);
 		sdlContext = nullptr;
 
@@ -313,6 +327,7 @@ namespace Renderer
 
 	unsigned int createTexture(const Texture::Type _type, const bool _linear, const bool _repeat, const unsigned int _width, const unsigned int _height, const void* _data)
 	{
+		//printf("CREATETEXTURE\n");//QNX
 		const GLenum type = convertTextureType(_type);
 		unsigned int texture;
 
@@ -354,6 +369,7 @@ namespace Renderer
 
 	void destroyTexture(const unsigned int _texture)
 	{
+		//printf("DESTROYTEXTURE\n");//QNX
 		GL_CHECK_ERROR(glDeleteTextures(1, &_texture));
 
 	} // destroyTexture
@@ -362,6 +378,7 @@ namespace Renderer
 
 	void updateTexture(const unsigned int _texture, const Texture::Type _type, const unsigned int _x, const unsigned _y, const unsigned int _width, const unsigned int _height, const void* _data)
 	{
+		//printf("UPDATETEXTURE\n");//QNX
 		const GLenum type = convertTextureType(_type);
 
 		GL_CHECK_ERROR(glBindTexture(GL_TEXTURE_2D, _texture));
@@ -395,6 +412,7 @@ namespace Renderer
 
 	void bindTexture(const unsigned int _texture)
 	{
+		//printf("BINDTEXTURE\n");//QNX
 		if(_texture == 0) GL_CHECK_ERROR(glBindTexture(GL_TEXTURE_2D, whiteTexture));
 		else              GL_CHECK_ERROR(glBindTexture(GL_TEXTURE_2D, _texture));
 
@@ -404,6 +422,7 @@ namespace Renderer
 
 	void drawLines(const Vertex* _vertices, const unsigned int _numVertices, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
 	{
+		//printf("DRAWLINES\n");//QNX
 		GL_CHECK_ERROR(glVertexAttribPointer(posAttrib, 2, GL_FLOAT,         GL_FALSE, sizeof(Vertex), (const void*)offsetof(Vertex, pos)));
 		GL_CHECK_ERROR(glVertexAttribPointer(texAttrib, 2, GL_FLOAT,         GL_FALSE, sizeof(Vertex), (const void*)offsetof(Vertex, tex)));
 		GL_CHECK_ERROR(glVertexAttribPointer(colAttrib, 4, GL_UNSIGNED_BYTE, GL_TRUE,  sizeof(Vertex), (const void*)offsetof(Vertex, col)));
@@ -419,6 +438,7 @@ namespace Renderer
 
 	void drawTriangleStrips(const Vertex* _vertices, const unsigned int _numVertices, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
 	{
+		//printf("QNX DEBUG: GLES20_DRTRISTRIPS\n"); //QNX
 		GL_CHECK_ERROR(glVertexAttribPointer(posAttrib, 2, GL_FLOAT,         GL_FALSE, sizeof(Vertex), (const void*)offsetof(Vertex, pos)));
 		GL_CHECK_ERROR(glVertexAttribPointer(texAttrib, 2, GL_FLOAT,         GL_FALSE, sizeof(Vertex), (const void*)offsetof(Vertex, tex)));
 		GL_CHECK_ERROR(glVertexAttribPointer(colAttrib, 4, GL_UNSIGNED_BYTE, GL_TRUE,  sizeof(Vertex), (const void*)offsetof(Vertex, col)));
@@ -434,6 +454,7 @@ namespace Renderer
 
 	void setProjection(const Transform4x4f& _projection)
 	{
+		//printf("SETPROJECTION\n");//QNX
 		projectionMatrix = _projection;
 
 		Transform4x4f mvpMatrix = projectionMatrix * worldViewMatrix;
@@ -445,6 +466,7 @@ namespace Renderer
 
 	void setMatrix(const Transform4x4f& _matrix)
 	{
+		//printf("SETMATRIX\n");//QNX
 		worldViewMatrix = _matrix;
 		worldViewMatrix.round();
 
@@ -457,6 +479,7 @@ namespace Renderer
 
 	void setViewport(const Rect& _viewport)
 	{
+		//printf("SETVIEWPORT\n");//QNX
 		// glViewport starts at the bottom left of the window
 		GL_CHECK_ERROR(glViewport( _viewport.x, getWindowHeight() - _viewport.y - _viewport.h, _viewport.w, _viewport.h));
 
@@ -466,6 +489,7 @@ namespace Renderer
 
 	void setScissor(const Rect& _scissor)
 	{
+		//printf("SETSCISSOR FUNCT \n"); //QNX
 		if((_scissor.x == 0) && (_scissor.y == 0) && (_scissor.w == 0) && (_scissor.h == 0))
 		{
 			GL_CHECK_ERROR(glDisable(GL_SCISSOR_TEST));
@@ -483,6 +507,7 @@ namespace Renderer
 
 	void setSwapInterval()
 	{
+		//printf("ATTEMPTING SWAP INTERVAL\n"); //QNX
 		// vsync
 		if(Settings::getInstance()->getBool("VSync"))
 		{
@@ -504,6 +529,7 @@ namespace Renderer
 
 	void swapBuffers()
 	{
+		printf("EmulationStation: | SWAP BUFFERS\n");//QNX
 		SDL_GL_SwapWindow(getSDLWindow());
 		GL_CHECK_ERROR(glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT));
 

From 4f4962a91e94ba4a81f413e124ef6bd569c20f96 Mon Sep 17 00:00:00 2001
From: Jai Moraes <jmoraes@blackberry.com>
Date: Thu, 20 Mar 2025 10:52:40 -0400
Subject: [PATCH 5/6] color fixed on theme side; now can use default
 convertColor

---
 es-core/src/renderers/Renderer.cpp        | 6 +-----
 es-core/src/renderers/Renderer_GLES20.cpp | 5 -----
 2 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/es-core/src/renderers/Renderer.cpp b/es-core/src/renderers/Renderer.cpp
index f469404d9a..14e724e3a1 100644
--- a/es-core/src/renderers/Renderer.cpp
+++ b/es-core/src/renderers/Renderer.cpp
@@ -272,12 +272,8 @@ namespace Renderer
 	void drawRect(const float _x, const float _y, /*const*/ float _w, /*const*/ float _h, const unsigned int _color, const unsigned int _colorEnd, bool horizontalGradient, const Blend::Factor _srcBlendFactor, const Blend::Factor _dstBlendFactor)
 	{
 		
-		//QNX
-		// _w = 1920;
-		// _h = 1080;
-		const unsigned int color    = /*convertColor(_color);*/ convertColor(0xAABBCCFF); //QNX
+		const unsigned int color    = convertColor(_color);
 		const unsigned int colorEnd = convertColor(_colorEnd);
-		printf("Draw rect call x%f y%f w%f h%f c=0x%08X \n", _x, _y, _w, _h, color); //QNX
 		Vertex             vertices[4];
 
 		vertices[0] = { { _x     ,_y      }, { 0.0f, 0.0f }, color };
diff --git a/es-core/src/renderers/Renderer_GLES20.cpp b/es-core/src/renderers/Renderer_GLES20.cpp
index 3e70bc9f90..4192e4d0ac 100644
--- a/es-core/src/renderers/Renderer_GLES20.cpp
+++ b/es-core/src/renderers/Renderer_GLES20.cpp
@@ -239,17 +239,12 @@ namespace Renderer
 
 	unsigned int convertColor(const unsigned int _color)
 	{
-		// convert from rgba to abgr
 		const unsigned char r = ((_color & 0xff000000) >> 24) & 255;
 		const unsigned char g = ((_color & 0x00ff0000) >> 16) & 255;
 		const unsigned char b = ((_color & 0x0000ff00) >>  8) & 255;
 		const unsigned char a = ((_color & 0x000000ff)      ) & 255;
 
-#ifdef __QNX__ //SHOULD BE RGBX
-		return ((r << 24) | (g << 16) | (b << 8) | (a));
-#else
 		return ((a << 24) | (b << 16) | (g << 8) | (r));
-	#endif
 
 	} // convertColor
 

From 68675d5dc0a1026cc71bd8403fc8a636e3d2cffc Mon Sep 17 00:00:00 2001
From: Jai Moraes <jmoraes@blackberry.com>
Date: Thu, 20 Mar 2025 14:59:44 -0400
Subject: [PATCH 6/6] Removing console spamming debugs

---
 es-app/src/main.cpp                       | 3 ---
 es-app/src/views/SystemView.cpp           | 2 --
 es-app/src/views/ViewController.cpp       | 3 ---
 es-core/src/Window.cpp                    | 1 -
 es-core/src/renderers/Renderer.cpp        | 9 ---------
 es-core/src/renderers/Renderer_GLES20.cpp | 8 +-------
 6 files changed, 1 insertion(+), 25 deletions(-)

diff --git a/es-app/src/main.cpp b/es-app/src/main.cpp
index 0d090fdfea..eb0b92cacf 100644
--- a/es-app/src/main.cpp
+++ b/es-app/src/main.cpp
@@ -291,7 +291,6 @@ void onExit()
 
 int main(int argc, char* argv[])
 {
-	printf("MAIN FUNC BEGIN\n");
 	std::locale::global(std::locale("C"));
 
 	if(!parseArgs(argc, argv))
@@ -354,7 +353,6 @@ int main(int argc, char* argv[])
 	CollectionSystemManager::init(&window);
 	MameNames::init();
 	window.pushGui(ViewController::get());
-	printf("main.cpp | Pushing GUI line 357\n"); //QNX
 
 	bool splashScreen = Settings::getInstance()->getBool("SplashScreen");
 
@@ -417,7 +415,6 @@ int main(int argc, char* argv[])
 		{
 			ViewController::get()->goToStart();
 		}else{
-			printf("main.cpp | Pushing GUI line 423\n");
 			window.pushGui(new GuiDetectDevice(&window, true, [] { ViewController::get()->goToStart(); }));
 		}
 	}
diff --git a/es-app/src/views/SystemView.cpp b/es-app/src/views/SystemView.cpp
index 8c3cf9b03a..5285958f8a 100644
--- a/es-app/src/views/SystemView.cpp
+++ b/es-app/src/views/SystemView.cpp
@@ -345,9 +345,7 @@ void SystemView::onCursorChanged(const CursorState& /*state*/)
 
 void SystemView::render(const Transform4x4f& parentTrans)
 {	
-	printf("SystemView.cpp | SystemView render call\n");
 	if(size() == 0){
-		printf("SystemView.cpp | systemview rendering exit due to failure\n");
 		return;  // nothing to render
 	}
 
diff --git a/es-app/src/views/ViewController.cpp b/es-app/src/views/ViewController.cpp
index 15f264ea64..7c0290c5d6 100644
--- a/es-app/src/views/ViewController.cpp
+++ b/es-app/src/views/ViewController.cpp
@@ -414,7 +414,6 @@ std::shared_ptr<IGameListView> ViewController::getGameListView(SystemData* syste
 
 std::shared_ptr<SystemView> ViewController::getSystemListView()
 {
-	//printf("System View Debug\n"); //QNX
 	//if we already made one, return that one
 	if(mSystemListView)
 		return mSystemListView;
@@ -462,7 +461,6 @@ void ViewController::update(int deltaTime)
 
 void ViewController::render(const Transform4x4f& parentTrans)
 {
-	printf("ViewController.cpp | ::render\n");
 	Transform4x4f trans = mCamera * parentTrans;
 	Transform4x4f transInverse;
 	transInverse.invert(trans);
@@ -497,7 +495,6 @@ void ViewController::render(const Transform4x4f& parentTrans)
 	// fade out
 	if(mFadeOpacity)
 	{
-		printf("Fade\n"); //QNX
 		unsigned int fadeColor = 0x00000000 | (unsigned char)(mFadeOpacity * 255);
 		Renderer::setMatrix(parentTrans);
 		Renderer::drawRect(0.0f, 0.0f, Renderer::getScreenWidth(), Renderer::getScreenHeight(), fadeColor, fadeColor);
diff --git a/es-core/src/Window.cpp b/es-core/src/Window.cpp
index 8b9c72f317..6a1afa505c 100644
--- a/es-core/src/Window.cpp
+++ b/es-core/src/Window.cpp
@@ -33,7 +33,6 @@ Window::~Window()
 
 void Window::pushGui(GuiComponent* gui)
 {
-	printf("Window.cpp: | Inside Window::pushGui <<<\n"); //QNX
 	if (mGuiStack.size() > 0)
 	{
 		auto& top = mGuiStack.back();
diff --git a/es-core/src/renderers/Renderer.cpp b/es-core/src/renderers/Renderer.cpp
index 14e724e3a1..fdb2574432 100644
--- a/es-core/src/renderers/Renderer.cpp
+++ b/es-core/src/renderers/Renderer.cpp
@@ -90,15 +90,6 @@ namespace Renderer
 		screenOffsetX = Settings::getInstance()->getInt("ScreenOffsetX") ? Settings::getInstance()->getInt("ScreenOffsetX") : 0;
 		screenOffsetY = Settings::getInstance()->getInt("ScreenOffsetY") ? Settings::getInstance()->getInt("ScreenOffsetY") : 0;
 		screenRotate  = Settings::getInstance()->getInt("ScreenRotate")  ? Settings::getInstance()->getInt("ScreenRotate")  : 0;
-		printf("Renderer.cpp | Reading in values... \n ");
-		printf("             | windowWidth %d \n ", windowWidth);
-		printf("             | windowHeight %d \n ", windowHeight);
-		printf("             | screenWidth %d \n ", screenWidth);
-		printf("             | screenHeight %d \n ", screenHeight);
-		printf("             | screenOffsetX %d \n ", screenOffsetX);
-		printf("             | screenOffsetY %d \n ", screenOffsetY);
-		printf("             | screenRotate %d \n ", screenRotate);
-		
 
 		setupWindow();
 
diff --git a/es-core/src/renderers/Renderer_GLES20.cpp b/es-core/src/renderers/Renderer_GLES20.cpp
index 4192e4d0ac..bc46e72af1 100644
--- a/es-core/src/renderers/Renderer_GLES20.cpp
+++ b/es-core/src/renderers/Renderer_GLES20.cpp
@@ -65,7 +65,6 @@ namespace Renderer
 		GL_CHECK_ERROR(glShaderSource(vertexShader, 1, &vertexSource, nullptr));
 		GL_CHECK_ERROR(glCompileShader(vertexShader));
 
-		printf("EmulationStation: | setupShaders\n");//QNX
 		{
 			GLint isCompiled = GL_FALSE;
 			GLint maxLength  = 0;
@@ -252,8 +251,7 @@ namespace Renderer
 
 	unsigned int getWindowFlags()
 	{
-		printf("EmulationStation: | WINDOW FLAGS\n");//QNX
-		return SDL_WINDOW_OPENGL; //is this right????? OPENGLES??
+		return SDL_WINDOW_OPENGL;
 
 	} // getWindowFlags
 
@@ -261,7 +259,6 @@ namespace Renderer
 
 	void setupWindow()
 	{
-		printf("EmulationStaion: | SETUP WINDOW\n");//QNX
 		SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK,  SDL_GL_CONTEXT_PROFILE_ES);
 		SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 2);
 		SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 0);
@@ -279,7 +276,6 @@ namespace Renderer
 
 	void createContext()
 	{
-		printf("EmulationStation: | CREATE CONTEXT\n");//QNX
 		sdlContext = SDL_GL_CreateContext(getSDLWindow());
 		SDL_GL_MakeCurrent(getSDLWindow(), sdlContext);
 
@@ -312,7 +308,6 @@ namespace Renderer
 
 	void destroyContext()
 	{
-		printf("EmulationStation: | DESTROY CONTEXT\n");//QNX
 		SDL_GL_DeleteContext(sdlContext);
 		sdlContext = nullptr;
 
@@ -524,7 +519,6 @@ namespace Renderer
 
 	void swapBuffers()
 	{
-		printf("EmulationStation: | SWAP BUFFERS\n");//QNX
 		SDL_GL_SwapWindow(getSDLWindow());
 		GL_CHECK_ERROR(glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT));
 
